ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-service-insertion.j2

```diff
@@ -7,4 +7,46 @@
 {% if router_service_insertion.enabled is arista.avd.defined(true) %}
 
 Router service-insertion is enabled.
+{%     if router_service_insertion.connections is arista.avd.defined %}
+
+## Connections
+{%         for connection in router_service_insertion.connections | arista.avd.natural_sort('name') %}
+{%             if connection.ethernet_interface.name is arista.avd.defined connection.ethernet_interface.next_hop is arista.avd.defined and connection.monitor_connectivity_host is arista.avd.defined %}
+
+### Connections Through Ethernet Interface
+
+| Name | Interface | Next Hop | Monitor Connectivity Host |
+| ---- | --------- | -------- | ------------------------- |
+{%                 break %}
+{%             endif %}
+{%         endfor %}
+{%         for connection in router_service_insertion.connections | arista.avd.natural_sort('name') %}
+{%             if connection.ethernet_interface.name is arista.avd.defined and connection.ethernet_interface.next_hop is arista.avd.defined and connection.monitor_connectivity_host is arista.avd.defined %}
+| {{ connection.name }} | {{ connection.ethernet_interface.name }} | {{ connection.ethernet_interface.next_hop }} | {{ connection.monitor_connectivity_host }} |
+{%             endif %}
+{%         endfor %}
+{%         for connection in router_service_insertion.connections %}
+{%             if connection.tunnel_interface.primary is arista.avd.defined and connection.tunnel_interface.secondary is arista.avd.defined and connection.monitor_connectivity_host is arista.avd.defined %}
+
+### Connections Through Tunnel Interface
+
+| Name | Primary Interface | Secondary Interface | Monitor Connectivity Host |
+| ---- | ----------------- | ------------------- | ------------------------- |
+{%                 break %}
+{%             endif %}
+{%         endfor %}
+{%         for connection in router_service_insertion.connections | arista.avd.natural_sort('name') %}
+{%             if connection.ethernet_interface is not arista.avd.defined %}
+{%                 if connection.tunnel_interface.primary is arista.avd.defined and connection.tunnel_interface.secondary is arista.avd.defined and connection.monitor_connectivity_host is arista.avd.defined %}
+| {{ connection.name }} | {{ connection.tunnel_interface.primary }} | {{ connection.tunnel_interface.secondary }} | {{ connection.monitor_connectivity_host }} |
+{%                 endif %}
+{%             endif %}
+{%         endfor %}
+{%     endif %}
```

Comment:
somthing like

**PLEASE** check this as everything is handtyped in Github.

Also notice that in some places I have added some default value ("-"). I am surprised your tests are passing as primary and secondary are not mandatory and you can have one without the other. Also there is not always a connectivity monitor.

```suggestion
{%         set ethernet_connections = [] %}
{%         set tunnel_connections = [] %}
{%         for connection in router_service_insertion.connections | arista.avd.natural_sort('name') %}
{%             if connection.ethernet_interface.name is arista.avd.defined connection.ethernet_interface.next_hop is arista.avd.defined and connection.monitor_connectivity_host is arista.avd.defined %}
{%               do ethernet_connections.append(connection) %}
{%             elif  connection.tunnel_interface.primary is arista.avd.defined or connection.tunnel_interface.secondary is arista.avd.defined %}
{%                do tunnel_connections.append(connection) %}
{%           endfor %}
{%           if ethernet_connections %}
### Connections Through Ethernet Interface

| Name | Interface | Next Hop | Monitor Connectivity Host |
| ---- | --------- | -------- | ------------------------- |
{%              for connection in ethernet_connections %}
| {{ connection.name }} | {{ connection.ethernet_interface.name }} | {{ connection.ethernet_interface.next_hop }} | {{ connection.monitor_connectivity_host | arista.avd.default("-") }} |
{%            endfor %}
{%         endif %}
{%           if tunnel_connections %}

### Connections Through Tunnel Interface

| Name | Primary Interface | Secondary Interface | Monitor Connectivity Host |
| ---- | ----------------- | ------------------- | ------------------------- |
{%              for connection in tunnel_connecitons %}
| {{ connection.name }} | {{ connection.tunnel_interface.primary | arista.avd.default("-") }} | {{ connection.tunnel_interface.secondary | arista.avd.default("-") }} | {{ connection.monitor_connectivity_host | arista.avd.default("-") }} |
{%            endfor %}
{%         endif %}
```

Reply:
Implemented this code and it works as expected. Added all possible scenarios in inventory.